plugins/woocommerce-blocks/tests/e2e/tests/product-gallery/inner-blocks/product-gallery-thumbnails/product-gallery-thumbnails.block_theme.spec.ts

```diff
@@ -1,418 +1,204 @@
-				.locator( '[data-type="woocommerce/product-gallery"]' )
-				.locator( '[data-type="core/group"]' )
-				.first();
+			).toBeVisible();
 
-			const groupBlockClassAttribute = await groupBlock.getAttribute(
-				'class'
-			);
-			expect( groupBlockClassAttribute ).toContain( 'is-layout-flex' );
-			expect( groupBlockClassAttribute ).toContain( 'is-nowrap' );
-
-			const isThumbnailsBlockEarlier = await editor.isBlockEarlierThan(
-				groupBlock,
-				'woocommerce/product-gallery-thumbnails',
-				'core/group'
-			);
-
-			expect( isThumbnailsBlockEarlier ).toBe( true );
-
-			await editor.saveSiteEditorEntities();
-
-			await page.goto( blockData.productPage );
-
-			const groupBlockFrontend = (
-				await frontendUtils.getBlockByClassWithParent(
-					'wp-block-group',
-					'woocommerce/product-gallery'
+			await expect(
+				productGalleryBlock.locator(
+					`[data-type="woocommerce/product-gallery-thumbnails"]:left-of(
+						[data-type="woocommerce/product-gallery-large-image"]
+					)`
 				)
-			 ).first();
-
-			const groupBlockFrontendClassAttribute =
-				await groupBlockFrontend.getAttribute( 'class' );
-			expect( groupBlockFrontendClassAttribute ).toContain(
-				'is-layout-flex'
-			);
-			expect( groupBlockFrontendClassAttribute ).toContain( 'is-nowrap' );
-
-			const isThumbnailsFrontendBlockEarlier =
-				await frontendUtils.isBlockEarlierThanGroupBlock(
-					groupBlockFrontend,
-					'woocommerce/product-gallery-thumbnails'
-				);
+			).toBeVisible();
 
-			expect( isThumbnailsFrontendBlockEarlier ).toBe( true );
+			await editor.saveSiteEditorEntities();
 		} );
 
-		// We can test the bottom position of thumbnails by cross-checking:
-		// - The Gallery block has the classes "is-layout-flex" and "is-vertical".
-		// - The Thumbnails block has a higher index than the Large Image block.
-		test( 'Position thumbnails on the bottom of the large image', async ( {
-			page,
-			editor,
-			frontendUtils,
-		} ) => {
-			// Currently we are adding the block under the legacy Product Image Gallery block, but in the future we have to add replace the product gallery block with this block.
-			const parentBlock = await editor.getBlockByName(
-				'woocommerce/product-image-gallery'
-			);
-			const clientId =
-				( await parentBlock.getAttribute( 'data-block' ) ) ?? '';
-			const parentClientId =
-				( await editor.getBlockRootClientId( clientId ) ) ?? '';
-
-			await editor.selectBlocks( parentBlock );
-			await editor.insertBlock(
-				{ name: 'woocommerce/product-gallery' },
-				{ clientId: parentClientId }
+		await test.step( 'in frontend', async () => {
+			await page.goto( '/product/v-neck-t-shirt/' );
+			const productGalleryBlock = page.locator(
+				'[data-block-name="woocommerce/product-gallery"]'
 			);
-			await ( await editor.getBlockByName( blockData.name ) ).click();
 
-			await editor.openDocumentSettingsSidebar();
-			await page
-				.locator(
-					blockData.selectors.editor.bottomPositionThumbnailsOption
```

Comment:
What are the benefits of moving this kind of selectors inline? I thought it was intentional to define all of them at the top, so if at some point the shape of the block changed, tests would be easier to update.

Reply:
Yup, it makes a lot of sense. Entirely agree that using user-facing attributes to locate elements is the way to go and hopefully we should have as few CSS selectors as possible. Thanks for the explanation, @WunderBart! :100: